docker-compose.dev.yml

```diff
@@ -0,0 +1,88 @@
+services:
+  astarte-housekeeping:
+    build:
+      target: builder
+    command: >
+      mix run -e Elixir.Astarte.Housekeeping.ReleaseTasks.init_database || exit 1
+      mix run -e Elixir.Astarte.Housekeeping.ReleaseTasks.migrate || exit 1
+      mix run --no-halt
+    develop:
+      watch:
+        - path: apps/astarte_housekeeping/lib
+          action: rebuild
+  astarte-housekeeping-api:
+    build:
+      target: builder
+    command: >
+      mix phx.server
+    develop:
+      watch:
+        - path: apps/astarte_housekeeping_api/lib
+          action: sync
+          target: /src/lib
```

Comment:
I'm not sure we want to trigger a reload only on `lib` changes. I would include also `mix.exs` and `mix.lock` so that a project setting or a library is modified, the change is watched.
The same goes for all other services